test/QueryToSocketDistributorTest.cpp

```diff
@@ -59,111 +63,142 @@ ASYNC_TEST(QueryToSocketDistributor, destructorRunsCleanup) {
 // _____________________________________________________________________________
 
 ASYNC_TEST(QueryToSocketDistributor, doubleSignalEndThrowsError) {
-  QueryToSocketDistributor queryToSocketDistributor{ioContext, noop};
-  co_await queryToSocketDistributor.signalEnd();
-  EXPECT_THROW(co_await queryToSocketDistributor.signalEnd(),
-               ad_utility::Exception);
+  auto ptr = std::make_shared<QueryToSocketDistributor>(ioContext, noop);
+  QueryToSocketDistributor& queryToSocketDistributor = *ptr;
```

Comment:
Why not just use `->` ?
Same for all the other cases in this file

Reply:
This was the minimal fix to make all your unit tests work:)
It's okay here and I would leave it as is, this one Extra line doesn't hurt too much.